### PR TITLE
Fix WinZip SFX folders not being scanned

### PR DESCRIPTION
**Pull request by TheRogueArchivist**
Use PKZIP extraction to fix WinZip SFX extraction not extracting folders.

### CODE REVIEW

File: BinaryObjectScanner/Packer/WinZipSFX.cs [hunk 1]
```diff
@@ -73,20 +73,21 @@ public class WinZipSFX : IExtractableNewExecutable, IExtractablePortableExecutab
         /// <summary>
         /// Handle common extraction between executable types
         /// </summary>
-        private static string? Extract(string file, bool includeDebug)
+        /// <inheritdoc/>
+        public string? Extract(string file, bool includeDebug)
         {
+            if (file == null)
+                return null;
```
Comment (mnadareski) [inline BinaryObjectScanner/Packer/WinZipSFX.cs:80]: These are unneeded as `file` is a non-nullable `string` in the parameters.

File: BinaryObjectScanner/Packer/WinZipSFX.cs [hunk 2]
```diff
@@ -73,20 +73,21 @@ public class WinZipSFX : IExtractableNewExecutable, IExtractablePortableExecutab
...
             {
-                // Should be using stream instead of file, but stream fails to extract anything. My guess is that the executable portion of the archive is causing stream to fail, but not file.
+                // Create a temp output directory
+                string tempPath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+                Directory.CreateDirectory(tempPath);
+
                 using (ZipArchive zipFile = ZipArchive.Open(file))
                 {
-                    if (!zipFile.IsComplete)
-                        return null;
```
Comment (mnadareski) [inline BinaryObjectScanner/Packer/WinZipSFX.cs:85]: Is there a reason that this was removed? Are there incomplete zipfiles that can be extracted?

Comment (TheRogueArchivist) [reply BinaryObjectScanner/Packer/WinZipSFX.cs:85]: I genuinely don't know, I ported the code from our PKZIP extraction, which doesn't seem to have this check. I figure this means that either this shouldn't have the check either, or the check should be added to our PKZIP extraction.

Comment (TheRogueArchivist) [reply BinaryObjectScanner/Packer/WinZipSFX.cs:85]: Though, I find it unlikely for there to be incomplete WinZip SFX files like this anyway, as as far as I know, they should always be completely self contained.

Comment (mnadareski) [reply BinaryObjectScanner/Packer/WinZipSFX.cs:85]: Please add this back as a minor bit of comfort.

**Review by mnadareski — APPROVED** [review]
Thanks

**Pull request merged**